server/src/server/read/advisory.rs

```diff
@@ -0,0 +1,48 @@
+use actix_multipart::form::tempfile::TempFile;
+use actix_multipart::form::MultipartForm;
+use actix_web::http::StatusCode;
+use actix_web::{post, web, HttpResponse, Responder};
+use anyhow::anyhow;
+use csaf::Csaf;
+use std::fs;
+
+use crate::server::Error;
+use crate::AppState;
+
+use sha2::{Digest, Sha256};
+use trustify_ingestors as ingestors;
+
+pub fn sha256(val: &str) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(val.as_bytes());
+    format!("{:x}", hasher.finalize())
+}
+
+#[derive(Debug, MultipartForm)]
+struct UploadForm {
+    #[multipart(rename = "file")]
+    files: Vec<TempFile>,
+}
+
+#[utoipa::path(responses((status = 200, description = "Upload a file")))]
+#[post("/advisories")]
+pub async fn upload_advisory(
+    state: web::Data<AppState>,
+    MultipartForm(form): MultipartForm<UploadForm>,
+) -> Result<impl Responder, Error> {
+    let temp_file = form.files.first().ok_or(Error::BadRequest {
```

Comment:
I really think we should avoid any filesystem interaction like that for uploading a file. I am also not sure if we should use multipart forms for uploading. What not follow the approach we used for streaming data: https://github.com/trustification/trustification/blob/0585db52cdcfd2ec4f58484777706b406b1ec40c/bombastic/api/src/server.rs#L317-L343

Reply:
I remove the multipart thing and replace it by `payload: web::Payload` just as we do in "trustification".